version: '3.8'

services:
  rabbitmq:
    image: rabbitmq:3.11-management
    hostname: rabbitmq
    networks:
      vpn_net:
        ipv4_address: 172.18.0.2
    environment:
      RABBITMQ_DEFAULT_USER: admin
      RABBITMQ_DEFAULT_PASS: admin
    ports:
      - "5672:5672"
      - "15672:15672"
    healthcheck:
      test: ["CMD", "rabbitmqctl", "status"]

  container_a:
    build:
      context: .
      dockerfile: container_a/Dockerfile
    cap_add:
      - NET_ADMIN
      - SYS_MODULE
    sysctls:
      net.ipv4.ip_forward: 1
    devices:
      - "/dev/net/tun:/dev/net/tun"
    networks:
      vpn_net:
        ipv4_address: 172.18.0.3
    volumes:
      - ./data:/app/data
      - ./logs:/var/log
    depends_on:
      rabbitmq:
        condition: service_healthy

  container_b:
    build:
      context: .
      dockerfile: container_b/Dockerfile
    cap_add:
      - NET_ADMIN
      - SYS_MODULE
    sysctls:
      net.ipv4.ip_forward: 1
    devices:
      - "/dev/net/tun:/dev/net/tun"
    networks:
      vpn_net:
        ipv4_address: 172.18.0.4
    volumes:
      - ./data:/app/data
      - ./logs:/var/log
    depends_on:
      rabbitmq:
        condition: service_healthy


networks:
  vpn_net:
    driver: bridge
    ipam:
      config:
        - subnet: 172.18.0.0/16  # Subnetz muss zur IP-Konfiguration passen
          gateway: 172.18.0.1
